codec/encode.go

```diff
@@ -167,3 +174,61 @@ func (se *Encoder) encodeTuple(t interface{}) (bytesEncoded int, err error) {
 
 	return bytesEncoded, nil
 }
+
+func (se *Encoder) encodeIntegerElements(arr []int) (bytesEncoded int, err error) {
+	var n int
+	n, err = se.encodeInteger(len(arr))
+	bytesEncoded += n
+
+	for _, elem := range arr {
+		n, err = se.encodeInteger(elem)
+		bytesEncoded += n
+	}
+
+	return bytesEncoded, err
+}
+
+// encodeVector encodes an interface where the underlying type is an array or slice
+// it writes the encoded length of the vector to the Encoder, then encodes and writes each value in the vector
+func (se *Encoder) encodeVector(t interface{}) (bytesEncoded int, err error) {
+	var n int
+	switch arr := t.(type) {
+	case []int:
+		n, err = se.encodeIntegerElements(arr)
+		bytesEncoded += n
+	case []*big.Int:
```

Comment:
The four following cases all use the same code except for the call to `se.encode...`

Reply:
unfortunately it must be this way due to the type switching. `arr` gets casted as the type in the case, if we put them all into one case it says `arr` is still an interface and doesn't compile